x-pack/plugins/aiops/public/components/log_rate_analysis/log_rate_analysis_options.tsx

```diff
@@ -0,0 +1,190 @@
+  defaultMessage: 'Fields',
+});
+const groupResultsOffMessage = i18n.translate(
+  'xpack.aiops.logRateAnalysis.resultsTable.groupedSwitchLabel.groupResultsOff',
+  {
+    defaultMessage: 'Off',
+  }
+);
+const groupResultsOnMessage = i18n.translate(
+  'xpack.aiops.logRateAnalysis.resultsTable.groupedSwitchLabel.groupResultsOn',
+  {
+    defaultMessage: 'On',
+  }
+);
+const resultsGroupedOffId = 'aiopsLogRateAnalysisGroupingOff';
+const resultsGroupedOnId = 'aiopsLogRateAnalysisGroupingOn';
+
+export interface LogRateAnalysisOptionsProps {
+  foundGroups: boolean;
+  growFirstItem?: boolean;
+}
+
+export const LogRateAnalysisOptions: FC<LogRateAnalysisOptionsProps> = ({
+  foundGroups,
+  growFirstItem = false,
+}) => {
+  const dispatch = useAppDispatch();
+
+  const { groupResults } = useAppSelector((s) => s.logRateAnalysis);
+  const { isRunning } = useAppSelector((s) => s.stream);
+  const fieldCandidates = useAppSelector((s) => s.logRateAnalysisFieldCandidates);
+  const { skippedColumns } = useAppSelector((s) => s.logRateAnalysisTable);
+  const { fieldFilterUniqueItems, initialFieldFilterSkippedItems } = fieldCandidates;
+  const fieldFilterButtonDisabled =
+    isRunning || fieldCandidates.isLoading || fieldFilterUniqueItems.length === 0;
+  const toggleIdSelected = groupResults ? resultsGroupedOnId : resultsGroupedOffId;
+
+  const onGroupResultsToggle = (optionId: string) => {
+    dispatch(setGroupResults(optionId === resultsGroupedOnId));
+    // When toggling the group switch, clear all row selections
+    dispatch(clearAllRowState());
+  };
+
+  const onVisibleColumnsChange = (columns: LogRateAnalysisResultsTableColumnName[]) => {
+    dispatch(setSkippedColumns(columns));
+  };
+
+  const onFieldsFilterChange = (skippedFieldsUpdate: string[]) => {
+    dispatch(setCurrentFieldFilterSkippedItems(skippedFieldsUpdate));
+  };
+
+  // Disable the grouping switch toggle only if no groups were found,
+  // the toggle wasn't enabled already and no fields were selected to be skipped.
+  const disabledGroupResultsSwitch = !foundGroups && !groupResults;
+
+  const toggleButtons = [
+    {
+      id: resultsGroupedOffId,
+      label: groupResultsOffMessage,
+      'data-test-subj': 'aiopsLogRateAnalysisGroupSwitchOff',
+    },
+    {
+      id: resultsGroupedOnId,
+      label: groupResultsOnMessage,
+      'data-test-subj': 'aiopsLogRateAnalysisGroupSwitchOn',
+    },
+  ];
+
+  return (
+    <>
+      <EuiFlexItem grow={growFirstItem}>
+        <EuiFlexGroup gutterSize="s" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <EuiText size="xs">{groupResultsMessage}</EuiText>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButtonGroup
+              data-test-subj={`aiopsLogRateAnalysisGroupSwitch${groupResults ? ' checked' : ''}`}
+              buttonSize="s"
+              isDisabled={disabledGroupResultsSwitch}
+              legend="Smart grouping"
```

Comment:
Does this text need to be internationalized?

Reply:
Updated in [7816dc5](https://github.com/elastic/kibana/pull/197943/commits/7816dc5e915b2f8c5b157cfa4256a131aebee003).